lib/classes/PluginManager.js

```diff
@@ -309,6 +310,53 @@ class PluginManager {
     });
   }
 
+  loadVariableResolvers(pluginInstance) {
+    const pluginName = pluginInstance.constructor.name;
+    for (const [variablePrefix, resolverOrOptions] of _.entries(
+      pluginInstance.variableResolvers || {}
+    )) {
+      let options = resolverOrOptions;
+      if (!options) {
+        this.serverless.cli.log(
+          `Warning! Ignoring falsey variableResolver for ${variablePrefix} in ${pluginName}.`
+        );
+        continue;
+      }
+      if (_.isFunction(resolverOrOptions)) {
+        options = {
+          resolver: resolverOrOptions,
+        };
+      }
+      if (!_.isObject(options)) {
+        throw new Error(
+          `Custom variable resolver {${variablePrefix}: ${JSON.stringify(
+            options
+          )}} defined by ${pluginName} isn't an object`
+        );
+      } else if (!variablePrefix.match(/[0-9a-zA-Z_-]/)) {
```

Comment:
I guess it should be `/^[0-9a-zA-Z_-]+$/`, as otherwise it matches only for existence of one of this chars

Reply:
yes. good catch.